### PR TITLE
Increase test coverage for NLU module

### LINKED ISSUE #6: Increase test coverage for NLU module
NLU module is being actively worked on so this is a priority for expanding test coverage.

It would be nice to get >90% coverage for the module.

### PR DESCRIPTION
**Pull request by IKostric**
Fixes #6

### CODE REVIEW

File: tests/nlu/test_rule_based_annotator.py [hunk 1]
```diff
@@ -20,9 +20,34 @@
...
+    "keywords": {
+        "a birthday party": "a birthday party",
+        "action figure": "action figure",
+    },
 }
 
 
+def process_value(text: str) -> str:
+    """Returns a process value fixture."""
+    return text
```
Comment (vinaysetty) [inline tests/nlu/test_rule_based_annotator.py:43]: not sure how returning same text is a process? is this supposed to be a dummy function?

File: tests/nlu/test_rule_based_annotator.py [hunk 2]
```diff
@@ -20,9 +20,34 @@
...
+
+
+def lematize_value(text: str) -> str:
+    """Returns a lematize value fixture."""
+    return text
```
Comment (vinaysetty) [inline tests/nlu/test_rule_based_annotator.py:48]: Is this supposed to mimic lemmatization?

**Review by vinaysetty — CHANGES_REQUESTED** [review]
Looks great! Just some minor nits.

File: moviebot/nlu/annotation/item_constraint.py [hunk 3]
```diff
@@ -75,6 +75,6 @@ def __str__(self) -> str:
         result = self.slot
 
         if self.value:
-            result += " " + opr + " " + str(self.value)
+            result += f" {opr} {str(self.value)}"
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlu/annotation/item_constraint.py:78]: Done

File: moviebot/nlu/annotation/rule_based_annotator.py [hunk 4]
```diff
@@ -101,21 +102,23 @@ def slot_annotation(
...
-                            p.value = f"{str(p.op)} {p.value}"
-                            p.op = Operator.EQ
-            else:
-                params = func(slot, user_utterance)
-        return params
+            return self._person_name_annotator(user_utterance)
+
+        func = getattr(self, f"_{slot}_annotator", None)
+        if not func:
+            return []
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlu/annotation/rule_based_annotator.py:110]: Done

File: moviebot/nlu/annotation/rule_based_annotator.py [hunk 5]
```diff
@@ -101,21 +102,23 @@ def slot_annotation(
...
+
+        constraints = func(slot, user_utterance)
+
+        # TODO https://github.com/iai-group/MovieBot/issues/142
+        # Remove this hack. Database lookup should be updated accordingly.
+        if slot == Slots.YEAR.value:
+            for c in constraints:
+                if c.op != Operator.EQ:
+                    c.value = f"{str(c.op)} {c.value}"
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlu/annotation/rule_based_annotator.py:119]: Done

File: moviebot/nlu/annotation/rule_based_annotator.py [hunk 6]
```diff
@@ -173,6 +176,7 @@ def _genres_annotator(
 
         if param:
             return [param]
+        return []
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlu/annotation/rule_based_annotator.py:179]: Done.

File: moviebot/nlu/annotation/rule_based_annotator.py [hunk 7]
```diff
@@ -360,12 +368,31 @@ def _person_name_annotator(
                         break
             if len(params) > 0:
                 return params
+        return []
+
+    def _get_digit_groups(self, token: Token) -> Tuple[str]:
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlu/annotation/rule_based_annotator.py:373]: Changed.

File: tests/nlu/test_rule_based_annotator.py (hunk 1 above)
Comment (IKostric) [reply tests/nlu/test_rule_based_annotator.py:43]: Added mock to the name to be more explicit.

File: tests/nlu/test_rule_based_annotator.py (hunk 2 above)
Comment (IKostric) [reply tests/nlu/test_rule_based_annotator.py:48]: Added mock to the name to be more explicit. The function is not relevant for the tests.

File: moviebot/nlu/annotation/rule_based_annotator.py [hunk 8]
```diff
@@ -223,9 +228,11 @@ def _title_annotator(
                     and len([int(val) for val in re.findall(r"\b\d+", gram)])
                     == 0
                 ):
-                    # check if
-                    # all words are in the list of stop words and no numbers
+                    # check if no words are in the list of stop words and
+                    # there are no numbers; This is used for partial matching.
```
Comment (WerLaj) [inline moviebot/nlu/annotation/rule_based_annotator.py:232]: Can we rephrase it a bit? Now it isn't very clear. Maybe something like: _Check if there are no words that are included in the list of stop words or that are numbers._

**Review by WerLaj — APPROVED** [review]
LGTM

File: tests/nlu/test_rule_based_annotator.py [hunk 9]
```diff
@@ -38,16 +39,22 @@
 }
 
 
-def process_value(text: str) -> str:
-    """Returns a process value fixture."""
+def mock_process_value(text: str) -> str:
```
Comment (vinaysetty) [inline tests/nlu/test_rule_based_annotator.py:42]: shouldn't this be marked @pytest.fixture if it is a fixture?

File: tests/nlu/test_rule_based_annotator.py [hunk 10]
```diff
@@ -38,16 +39,22 @@
...
+    """Returns a mock process value fixture."""
     return text
 
 
-def lematize_value(text: str) -> str:
-    """Returns a lematize value fixture."""
+def mock_lematize_value(text: str) -> str:
```
Comment (vinaysetty) [inline tests/nlu/test_rule_based_annotator.py:47]: Same as above.

**Review by vinaysetty — APPROVED** [review]
LGTM! Just minor see about pytest fixture and resolve it yourself.

File: moviebot/nlu/annotation/rule_based_annotator.py (hunk 8 above)
Comment (IKostric) [reply moviebot/nlu/annotation/rule_based_annotator.py:232]: Done

File: tests/nlu/test_rule_based_annotator.py (hunk 9 above)
Comment (IKostric) [reply tests/nlu/test_rule_based_annotator.py:42]: Fixed docstring

**Pull request merged**